pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -664,6 +670,11 @@ func translateHeaderMatch(name string, in *networking.StringMatch, node *model.P
 		Name: name,
 	}
 
+	if in == nil {
```

Comment:
can you check if crd validation allows nil ?

Reply:
Oh, CRD validation does not permit "nil" value.
Could you guide a way to make CRD validation allow nil?
(protobuf seems not to provide a explicit way to allow nil value for map)

Reply:
Here is result. Currently, there seems no ways to allow nil to a value part in key-value of a map in protobuf. To express the presence of a header, I think it would be better to use the Regex at this moment, and think about making a StringMatch nullable in HeaderMatch.

Reply:
Even if CRD does not permit `nil`, guarding for `nil` value still seems necessary, so I would like to keep that part.